aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -332,6 +407,272 @@ def _build_middleware_stack(self, router_middlewares: List[Callable[..., Any]])
 
         self._middleware_stack_built = True
 
+    @property
+    def dependant(self) -> "Dependant":
+        if self._dependant is None:
```

Comment:
Memorize for performance. I couldn't find an easy way to memoize a property that works with python 3.7